image-encapsulate.cpp

```diff
@@ -2,13 +2,12 @@
 #include "brightener.h"
 
 int main() {
-	auto image = std::make_unique<Image>(512, 512);
+	auto image = std::make_shared<Image>(512, 512);
 	std::cout << "Brightening a 512 x 512 image\n";
 	ImageBrightener brightener(std::move(image));
-	if (brightener.ValidateImage()) {
+	if (image->ValidateImage()) {
```

Comment:
Resource usage can be further optimized - in case the Image is not valid, no need to instantiate `ImageBrightener` itself (line 7 can move inside the `if` block)